src/components/Input.js

```diff
@@ -1,32 +1,69 @@
-import { useTransition, animated } from 'react-spring'
+import { useTransition, animated } from "react-spring";
 
-const Input = ({id, className, placeholder, type="text", value, isValid, setIsValid, 
-    user, setUser, invMsg, invMsgDirection, onChange}) => {
-
-    const transition = useTransition( !isValid[`${id}`], {
-        from: { opacity: 0},
-        enter: { opacity: 1},
-        leave: { opacity: 0},
-        config: {duration: 600}
-    } )
+const Input = ({
+    id,
+    className,
+    placeholder,
+    type = "text",
+    value,
+    isValid,
+    setIsValid,
+    user,
+    setUser,
+    invMsg,
+    invMsgDirection,
+    onChange,
+    required = true,
+}) => {
+    const transition = useTransition(!isValid[`${id}`], {
+        from: { opacity: 0 },
+        enter: { opacity: 1 },
+        leave: { opacity: 0 },
+        config: { duration: 600 },
+    });
 
     return (
         <div className="holder">
-            <input id={id} type={type}
-                className={`${className} ${isValid[`${id}`]?"":"invalid-input"}`}
-                placeholder={placeholder} value={value === undefined ? user[`${id}`] : value} 
-                onChange={onChange==undefined ? event => setUser({...user, [`${id}`]: event.target.value}): onChange}
+            <input
+                id={id}
+                type={type}
+                className={`form-input ${className} ${
+                    isValid[`${id}`] ? "" : "invalid-input"
+                }`}
+                placeholder={placeholder}
+                value={value === undefined ? user[`${id}`] : value}
+                onChange={
+                    onChange == undefined
```

Comment:
jak nie przyszedł prop onChange to nie robiłbym domyślnie żeby obsługiwał setUser, bo nie będzie się dało użyć gdzie indziej inputu.